contracts/cw721-expiration/src/execute.rs

```diff
@@ -1,282 +1,225 @@
-use cosmwasm_std::{
-    Addr, Binary, Coin, Deps, DepsMut, Empty, Env, MessageInfo, Response, StdResult, Storage,
-};
-use cw721::{Cw721Execute, Expiration};
-use cw721_base::Cw721Contract;
-
 use crate::{
-    error::ContractError, msg::ExecuteMsg, msg::InstantiateMsg, state::Cw721ExpirationContract,
-    Extension,
+    error::ContractError, msg::InstantiateMsg, state::Cw721ExpirationContract, CONTRACT_NAME,
+    CONTRACT_VERSION,
+};
+use cosmwasm_std::{Binary, CustomMsg, DepsMut, Env, MessageInfo, Response};
+use cw721_base::{
+    msg::{Cw721ExecuteMsg, Cw721InstantiateMsg},
+    traits::{
+        Contains, Cw721CustomMsg, Cw721Execute, Cw721State, FromAttributesState, StateFactory,
+        ToAttributesState,
+    },
+    Expiration,
 };
-use cw721_base::InstantiateMsg as Cw721InstantiateMsg;
 
-impl<'a> Cw721ExpirationContract<'a> {
+impl<
+        'a,
+        TNftExtension,
+        TNftExtensionMsg,
+        TCollectionExtension,
+        TCollectionExtensionMsg,
+        TExtensionMsg,
+        TExtensionQueryMsg,
+        TCustomResponseMsg,
+    >
+    Cw721ExpirationContract<
+        'a,
+        TNftExtension,
+        TNftExtensionMsg,
+        TCollectionExtension,
+        TCollectionExtensionMsg,
+        TExtensionMsg,
+        TExtensionQueryMsg,
+        TCustomResponseMsg,
+    >
+where
+    TNftExtension: Cw721State + Contains,
+    TNftExtensionMsg: Cw721CustomMsg + StateFactory<TNftExtension>,
+    TCollectionExtension: Cw721State + ToAttributesState + FromAttributesState,
+    TCollectionExtensionMsg: Cw721CustomMsg + StateFactory<TCollectionExtension>,
+    TExtensionQueryMsg: Cw721CustomMsg,
+    TCustomResponseMsg: CustomMsg,
+{
```

Comment:
Sylvia solves this. :joy:

Would be good to add a template for custom `cw721` contracts, can update #120 after this PR.